### ISSUE TITLE
ana

### ISSUE THREAD

**Issue by YoshidaHiroya** [YoshidaHiroya/kyotoijuku#1]
import pandas as pd
import seaborn as sns
import statsmodels.api as sm
from sklearn.metrics import classification_report
df=pd.read_csv('result.csv')
df.head(6)

df2 = df.dropna(how='any').dropna(how='any', axis=1)
df2.head(10)

from sklearn.model_selection import train_test_split
x=df2.loc[:,'E_d':]
t=df2['result']

x_train,x_val,y_train,y_val=train_test_split(x,t,test_size=0.2,random_state=0)
from sklearn.linear_model import LogisticRegression

model=LogisticRegression(random_state=0,C=0.1,multi_class='auto',solver='lbfgs')

model.fit(x_train,y_train)
print(model.score(x_train,y_train))
model.score(x_val,y_val)

from sklearn.metrics import confusion_matrix
pred = model.predict(x_val)
cm = confusion_matrix(y_val, pred)
sns.heatmap(cm, annot=True, cmap='Oranges', fmt='d')

pred=model.predict(x_val)
out_put=classification_report(y_pred=pred,y_true=y_val)
print(out_put)

from sklearn.metrics import roc_curve, roc_auc_score
import matplotlib.pyplot as plt
from IPython.display import display
from sklearn.datasets import load_breast_cancer
colors = plt.rcParams['axes.prop_cycle'].by_key()['color']
proba_lr = model.predict_proba(x_val)[:, 1]  # ロジスティック回帰モデルで確率を出力

# ROC曲線を描写
fpr, tpr, thresholds = roc_curve(y_val, proba_lr)
plt.plot(fpr, tpr, color=colors[0], label='logistic')
plt.fill_between(fpr, tpr, 0, color=colors[0], alpha=0.1)

plt.xlabel('FPR')
plt.ylabel('TPR')
plt.legend()
plt.show()

# AUCの計算
print(f'ロジスティック回帰モデル AUR: {roc_auc_score(y_val, proba_lr):.4f}')
